automations/python/label_pr.py

```diff
@@ -76,21 +80,54 @@ def get_pull_request(gh: Github, html_url: str) -> PullRequest:
     return gh.get_organization(org).get_repo(repo).get_pull(pr_num)
 
 
+def get_authenticated_html(url: str) -> str:
+    """
+    Login to the GitHub UI using the username and password, followed by 2FA. Then
+    navigate to the specified URL as the authenticated user and scrape the text body.
+
+    :param url: the URL to scrape after authenticating in GitHub
+    :return: the text content of the scraped page
+    """
+
+    browser = mechanize.Browser()
+    browser.set_cookiejar(CookieJar())
+    browser.open("https://github.com/login/")
+    browser.select_form(nr=0)
+    browser.form["login"] = os.getenv("GH_LOGIN")
+    browser.form["password"] = os.getenv("GH_PASSWORD")
+    browser.submit()
+
+    browser.select_form(nr=0)
+    browser.form["otp"] = pyotp.TOTP(os.getenv("GH_2FA_SECRET")).now()
```

Comment:
My [League of Legends](https://www.leagueoflegends.com/en-us/) days are showing, I keep reading `otp` in my head as "one trick pony" 😂 🐴